terraform-docker/apply/main.yaml

```diff
@@ -52,6 +61,55 @@ stages:
                         artifact: "${{ env.name }}.enc"
                         path: $(Build.Repository.LocalPath)/${{ parameters.terraformFolder }}/.terraform/plans
 
+                    - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+                      displayName: "Get AWS credentials"
+                      inputs:
+                        workingDirectory: "."
+                        awsCredentials: ${{ format(parameters.awsServiceConnectionTemplate, env.name) }}
+                        regionName: ${{ parameters.awsRegion }}
+                        scriptType: inline
+                        inlineScript: |
+                          set -ex
+                          AWS_CONFIG_FILE=${HOME}/.aws/config
+                          AWS_SHARED_CREDENTIALS_FILE=${HOME}/.aws/credentials
+                          TMPFILE=${HOME}/.aws/tmp.json
+                          mkdir $(dirname $TMPFILE)
+                          aws sts get-federation-token --name xkf-$ENV --policy-arns arn="$AWS_ARN" --region $AWS_REGION > $TMPFILE
+
+                          KEY_ID=$(cat $TMPFILE |jq -r .Credentials.AccessKeyId)
+                          KEY_SECRET=$(cat $TMPFILE |jq -r .Credentials.SecretAccessKey)
+                          SESSION_TOKEN=$(cat $TMPFILE |jq -r .Credentials.SessionToken)
+
+                          # Create the credentials file from AWS session token
+                          cat <<EOF > $AWS_SHARED_CREDENTIALS_FILE
+                          [xkf-$ENV]
+                          aws_access_key_id = $KEY_ID
+                          aws_secret_access_key = $KEY_SECRET
+                          aws_session_token = $SESSION_TOKEN
+
+                          EOF
+
+                          # Create the config file
+
+                          cat <<EOF > $AWS_CONFIG_FILE
+                          [default]
+                          region = $AWS_REGION
+                          output = json
+
+                          [profile xkf-$ENV]
+                              source_profile = xkf-$ENV
+                              role_arn = $AWS_ARN
+
+                          EOF
+
+                          sudo chown -R 1000:1000 ${AWS_CONFIG_FILE}
+                          sudo chown -R 1000:1000 ${AWS_SHARED_CREDENTIALS_FILE}
```

Comment:
Do we want to move this chown down to the Azure section?
The reason that i keept it up here is that I want to mix the two environments as little as possible.
Could easily add a if for it down in the Azure section.